sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -0,0 +1,253 @@
+     */
+    private Mono<java.net.http.HttpRequest> toJdkHttpRequest(HttpRequest request) {
+        return Mono.fromCallable(() -> {
+            final java.net.http.HttpRequest.Builder builder = java.net.http.HttpRequest.newBuilder();
+            try {
+                builder.uri(request.getUrl().toURI());
+            } catch (URISyntaxException e) {
+                throw Exceptions.propagate(e);
+            }
+            final HttpHeaders headers = request.getHeaders();
+            if (headers != null) {
+                for (HttpHeader header : headers) {
+                    final String headerName = header.getName();
+                    if (!JDK12_RESTRICTED_HEADERS.contains(headerName)) {
+                        final String headerValue = header.getValue();
+                        builder.setHeader(headerName, headerValue);
+                    } else {
+                        logger.logExceptionAsError(
+                            new IllegalArgumentException("The header " +
+                                "'" + headerName + "' is restricted by default in JDK HttpClient 12 and above." +
+                                "(unless it is whitelisted in JAVA_HOME/conf/net.properties)"));
+                    }
+                }
+            }
+            switch (request.getHttpMethod()) {
+                case GET:
+                    return builder.GET().build();
+                case HEAD:
+                    return builder.method("HEAD", noBody()).build();
+                default:
+                    final String contentLength = request.getHeaders().getValue("content-length");
+                    final BodyPublisher bodyPublisher = toBodyPublisher(request.getBody(), contentLength);
+                    return builder.method(request.getHttpMethod().toString(), bodyPublisher).build();
+            }
+        });
+    }
+
+    /**
+     * Create BodyPublisher from the given java.nio.ByteBuffer publisher.
+     *
+     * @param bbPublisher stream of java.nio.ByteBuffer representing request content
+     * @return the request BodyPublisher
+     */
+    private static BodyPublisher toBodyPublisher(Flux<ByteBuffer> bbPublisher, String contentLength) {
+        if (bbPublisher == null) {
+            return noBody();
+        }
+        final Flow.Publisher<ByteBuffer> bbFlowPublisher = JdkFlowAdapter.publisherToFlowPublisher(bbPublisher);
+        if (CoreUtils.isNullOrEmpty(contentLength)) {
+            return fromPublisher(bbFlowPublisher);
+        } else {
+            long contentLengthLong = Long.parseLong(contentLength);
+            if (contentLengthLong < 1) {
+                return fromPublisher(bbFlowPublisher);
+            } else {
+                return fromPublisher(bbFlowPublisher, contentLengthLong);
+            }
+        }
+    }
+
+    /**
+     * Get the java runtime major version.
+     *
+     * @return the java major version
+     */
+    private int getJavaVersion() {
```

Comment:
This could be made static right?

Reply:
could be, it uses instance-level logger, so static means we need to pass it.

Reply:
going with a private instance-level method, Can be static if we ever move this method to utils.